api/envoy/extensions/original_ip_detection/custom_header/v3/custom_header.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+package envoy.extensions.original_ip_detection.custom_header.v3;
+
+import "envoy/extensions/original_ip_detection/common/v3/original_ip_detection.proto";
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.original_ip_detection.custom_header.v3";
+option java_outer_classname = "CustomHeaderProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Custom header original IP detection extension]
+
+// This extension allows for the original downstream remote IP to be detected
+// by reading the value for a configured header name. If the value is successfully parsed
+// as an IP, it'll be treated as the effective downstream remote address and seen as such
+// by all filters. If the detection or parsing of the header value fails, Envoy will fallback
+// to its default detection extension.
```

Comment:
This seems odd given all of the error params that are in common config? Do you instead want a repeated list of typed_configs in the HCM config to handle fallback? That seems better?

Reply:
Hold on, I _think_ you are saying two things:

1) the error params are common config and thus should be shared across extension? What part is odd? To which I'd disagree, I think different extensions can have different ways of handling rejection or not...

and:

2) The repeated list of extensions would be for users to be able to handle their own fallback? We'd still need the default extension, not declared in that list right?